### PR TITLE
Turn IAP constants into properties

### LINKED ISSUE #1192: Make IAP settings configurable properties
This is a follow up on #1016 and PR #1190.

These should become properties:
* PUBLIC_KEY_VERIFICATION_LINK
* ENCRYPTION_ALGORITHM
* IAP_HEADER

### PR DESCRIPTION
**Pull request by elefeint**
I added a test for one property; not sure if it's worth adding the rest. Also renamed a test constant to reflect its purpose better.

Fixes #1192.

### CODE REVIEW

File: spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc [hunk 1]
```diff
@@ -35,6 +35,23 @@ dependencies {
 }
 ----
 
+== Customizable Properties
+The following properties are available with `spring.cloud.gcp.security.iap` prefix:
 
+|===
+|Name |Value |Description
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc]: We normally have four columns: Name, Description, Required, Default value.

File: spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc [hunk 2]
```diff
@@ -35,6 +35,23 @@ dependencies {
 }
 ----
 
+== Customizable Properties
+The following properties are available with `spring.cloud.gcp.security.iap` prefix:
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc]: There should probably be a some language about not modifying these for production use and only using for testing.

File: spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc [hunk 3]
```diff
@@ -35,6 +35,23 @@ dependencies {
 }
 ----
 
+== Customizable Properties
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc]:
```suggestion
== Configuration
```

File: spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc (hunk 1 above)
Comment (elefeint) [reply spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc]: What happens if it's required, but a default is provided in autoconfig (so, for end user, overriding is not required)?
Also, should I enforce required non-null with JSR303 annotations or something?

File: spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc [hunk 4]
```diff
@@ -35,6 +35,28 @@ dependencies {
...
 
+CAUTION: Modifying registry, algorithm, and header properties might be useful for testing, but the defaults should not be changed in production.
+
+|===
+|Name |Description |Required |Default
+
+|`spring.cloud.gcp.security.iap.registry`
+|Link to JWK public key registry.
+|true
+|https://www.gstatic.com/iap/verify/public_key-jwk
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc]:
```suggestion
|`https://www.gstatic.com/iap/verify/public_key-jwk`
```

File: spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc [hunk 5]
```diff
@@ -35,6 +35,28 @@ dependencies {
...
+
+|`spring.cloud.gcp.security.iap.algorithm`
+|Encryption algorithm used to sign the JWK token.
+|true
+|ES256
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc]:
```suggestion
|`ES256`
```

File: spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc [hunk 6]
```diff
@@ -35,6 +35,28 @@ dependencies {
...
+
+|`spring.cloud.gcp.security.iap.header`
+|Header from which to extract the JWK key.
+|true
+|x-goog-iap-jwt-assertion
```
Comment (meltsufin) [inline spring-cloud-gcp-docs/src/main/asciidoc/security-iap.adoc]:
```suggestion
|`x-goog-iap-jwt-assertion`
```

**Comment by elefeint** [comment]
@artembilan What do you think?

**Pull request merged**